### PR TITLE
Multisig ISM

### PR DESCRIPTION
**Pull request by asaj**

### CODE REVIEW

File: HIP-3.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+bytes32 public commitment;
+
+function verify(bytes calldata _metadata, bytes calldata _message)
+    external
+    returns (bool)
+{
+    uint8 _metadataThreshold = _metadata.threshold();
+    address[] memory _metadataValidators = _metadata.validators();
+
+    bytes32 _metadataCommitment = keccak256
```
Comment (tkporter) [inline HIP-3.md:68]: Think this line is unintentional

File: HIP-3.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+### **Rationale**
+
+#### \_message as an argument to validatorsAndThreshold()
+
+Passing `_message` as argument to `validatorsAndThreshold()` allows for a broad array of customizability.
+
+This allows users to create `MultisigIsms` that are capable of verifying messages from many remote domains (by switching on `message.origin()`).
+
+It also allows users to create `MultisigIsms` that are aware of the semantic meaning of the interchain message, allowing them to e.g. increase or decrease the threshold based on message value.
```
Comment (tkporter) [inline HIP-3.md:105]: This is really cool

**Review by tkporter — COMMENTED** [review]
LGTM- I'm still digesting the greater context of this, but so far I agree with everything in here

File: HIP-3.md (hunk 1 above)
Comment (asaj) [reply HIP-3.md:68]: Indeed, good catch, will remove

**Review by nambrot — APPROVED** [review]
Nice and concise HIP as expected for this author :)

**Pull request merged**